File: Projet_LO52/lo52_project_v2/app/src/main/java/com/example/lo52_project_v2/model/dao/EquipeDao.java
package com.example.lo52_project_v2.model.dao;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.example.lo52_project_v2.model.bo.Equipe;

import java.util.List;

@Dao
public interface EquipeDao {
    @Insert
    public long insertEquipe(Equipe equipe);

    @Query("SELECT * FROM Equipe")
    List<Equipe> getAll();

    @Query("SELECT * FROM Equipe WHERE idCourse LIKE :idCourse ")
    List<Equipe> getCorrespondingTeams(int idCourse);

    @Update
    public void update(Equipe team);

    @Update
    public void update(Equipe... teams);

    @Update
    public void update(List<Equipe> teams);
}

File: Projet/GestionCourse/Gestion-Course/app/src/main/java/com/example/gestion_course/dao/EquipeDao.kt
package com.example.gestion_course.dao

import androidx.room.*
import com.example.gestion_course.entities.Equipe
import com.example.gestion_course.entities.EquipeAvecParticipants
import com.example.gestion_course.entities.Participant

@Dao
interface EquipeDao {
    @Query("SELECT * FROM equipe")
    fun getAll(): List<Equipe>

    @Transaction
    @Query("SELECT * FROM Equipe INNER JOIN Participant WHERE Equipe.num_equipe = Participant.num_equipe_participant GROUP BY num_equipe_participant")
    fun getEquipeAvecParticipants(): List<EquipeAvecParticipants>

    @Insert
    suspend fun insertAll(vararg equipes: Equipe)

    @Insert
    suspend fun insertEquipeAvecParticipants(equipe: Equipe, participants: List<Participant>)

    @Insert
    fun insertParticipants(participants: List<Participant>)

    @Delete
    fun delete(equipe: Equipe)

}